openedx/core/djangoapps/content/learning_sequences/api/processors/enrollment_track_partition_groups.py

```diff
@@ -0,0 +1,61 @@
+# lint-amnesty, pylint: disable=missing-module-docstring
+import logging
+
+from common.lib.xmodule.xmodule.partitions.enrollment_track_partition_generator import (
+    create_enrollment_track_partition_with_course_id
+)
+from common.lib.xmodule.xmodule.partitions.partitions import (
+    ENROLLMENT_TRACK_PARTITION_ID,
+)
+from common.lib.xmodule.xmodule.partitions.partitions_service import get_user_partition_groups
+
+from .base import OutlineProcessor
+
+log = logging.getLogger(__name__)
+
+
+class EnrollmentTrackPartitionGroupsOutlineProcessor(OutlineProcessor):
+    """
+    Processor responsible for applying all enrollment track user partition group to the outline.
+
+    Confining the processor to only EnrollmentTrack user partition is a significant limitation.
+    Nonetheless, it is a step towards the goal of supporting
+    all partition schemes in the future
+    """
+    def __init__(self, course_key, user, at_time):
+        super().__init__(course_key, user, at_time)
+        self.user_groups = []
+        self.enrollment_track_groups = {}
+
+    def load_data(self):
+        user_partition = create_enrollment_track_partition_with_course_id(self.course_key)
+        self.enrollment_track_groups = get_user_partition_groups(self.course_key, [user_partition], self.user)
+
+    def _should_block_be_removed(self, block):
```

Comment:
Tiny (optional) nit: You're passing in the `block` here, but the only thing it needs to make its decisions is `block.user_partition_groups`. You could pass that in instead, which has the small bonus of only then passing one type of object to this method, instead of passing both sections and sequences.